src/shogun/kernel/GaussianShiftKernel.cpp

```diff
@@ -44,12 +41,11 @@ CGaussianShiftKernel::~CGaussianShiftKernel()
 float64_t CGaussianShiftKernel::compute(int32_t idx_a, int32_t idx_b)
 {
 	int32_t alen, blen;
-	bool afree, bfree;
 
-	float64_t* avec=
-		((CDenseFeatures<float64_t>*) lhs)->get_feature_vector(idx_a, alen, afree);
-	float64_t* bvec=
-		((CDenseFeatures<float64_t>*) rhs)->get_feature_vector(idx_b, blen, bfree);
+	SGVector<float64_t> avec=((CDotFeatures *)lhs)->get_computed_dot_feature_vector(idx_a);
+	SGVector<float64_t> bvec=((CDotFeatures *)rhs)->get_computed_dot_feature_vector(idx_b);
+	alen=avec.vlen;
```

Comment:
I added an entrance task on this, see
https://github.com/shogun-toolbox/shogun/issues/2702